iceoryx_posh/source/roudi/service_registry.cpp

```diff
@@ -96,19 +97,21 @@ void ServiceRegistry::remove(const capro::ServiceDescription& serviceDescription
     {
         removeIndexFromMap(m_serviceMap, index);
         removeIndexFromMap(m_instanceMap, index);
+        removeIndexFromMap(m_eventMap, index);
     }
 }
 
 void ServiceRegistry::find(ServiceDescriptionVector_t& searchResult,
                            const cxx::optional<capro::IdString_t>& service,
-                           const cxx::optional<capro::IdString_t>& instance) const noexcept
+                           const cxx::optional<capro::IdString_t>& instance,
+                           const cxx::optional<capro::IdString_t>& event) const noexcept
 {
-    cxx::vector<uint64_t, MAX_SERVICE_DESCRIPTIONS> intersection;
-
+    ServiceDescriptionVector_t serviceInstanceResult;
     // Find (K1, K2)
```

Comment:
Now that event is added could we please remove those runtime big O notation comments.

Reply:
I didn't check the O notation for correctness, but the branches where it's noted didn't change. Therefore, no updates needed now. Should be updated (or removed) when the prefix tree is integrated.

Reply:
I will update this/remove as needed.